src/ess/reduce/ui.py

```diff
@@ -18,145 +20,242 @@
-)
-
-run_button = widgets.Button(
-    description='Run',
-    disabled=True,
-    button_style='success',
-    tooltip='Run',
-)
+class OutputSelectionWidget(widgets.VBox):
+    def __init__(self, workflow: sl.Pipeline, **kwargs):
+        self.typical_outputs_widget = widgets.SelectMultiple(
+            options=get_typical_outputs(workflow),
+            layout=Layout(width='90%', height='auto'),
+        )
+        self.possible_outputs_widget = widgets.SelectMultiple(
+            options=get_possible_outputs(workflow),
+            layout=Layout(width='90%', height='auto'),
+        )
+        # Wrapping the selection widget in an individual accordion.
+        # It is not possible to have multiple sections open in one ``Accordion`` widget.
+        _typical_selection = _wrap_foldable(
+            self.typical_outputs_widget, title='Typical Outputs'
+        )
+        _typical_selection.selected_index = 0  # Open typical outputs by default
+        _possible_selection = _wrap_foldable(
+            self.possible_outputs_widget, title='Extended Outputs'
+        )
+        super().__init__([_typical_selection, _possible_selection], **kwargs)
 
-output = widgets.Output()
+    @property
+    def value(self) -> set[Key]:
+        return set(
+            self.typical_outputs_widget.value + self.possible_outputs_widget.value
+        )
 
 
-def collect_values(
-    parameter_box: widgets.VBox, param_keys: Iterable[Key]
-) -> dict[Key, Any]:
-    return {
-        node: parameter_box.children[i].children[0].value
-        for i, node in enumerate(param_keys)
-        if (
-            not isinstance(
-                widget := parameter_box.children[i].children[0], SwitchWidget
+class ParameterBox(widgets.VBox):
+    def __init__(
+        self,
+        registry_getter: Callable[[], dict[Key, Parameter]],
+        **kwargs,
+    ):
+        self.parameter_refresh_button = widgets.Button(
+            description='Refresh Parameters',
+            disabled=False,
+            button_style='success',
+            tooltip='Generate Parameter Input Widgets',
+        )
+        self._input_registry = {}
+        self._input_widgets = {}
+        self._input_box = widgets.VBox()
+
+        def _refresh_input_box(_: widgets.Button):
+            new_input_parameters = registry_getter()
+            self._input_registry.clear()
+            self._input_registry.update(new_input_parameters)
+            self._input_widgets.clear()
+            self._input_widgets.update(
+                {
+                    node: widgets.HBox([create_parameter_widget(parameter)])
+                    for node, parameter in registry_getter().items()
+                }
             )
+            self._input_box.children = list(self._input_widgets.values())
+
+        self.parameter_refresh_button.on_click(_refresh_input_box)
+
+        super().__init__([self.parameter_refresh_button, self._input_box], **kwargs)
+
+    def collect_values(self) -> dict[Key, Any]:
```

Comment:
(minor) Is there a reason we don't want to call this just `values` instead of `collect_values`?

Reply:
Just wanted to make it explicit.
But since all the wrapper widget has `value` property, it might be okay to make a `value` property...?

And `values` might be confusing since you might expect it to work like `dict.values`.

Reply:
I updated it to be a `value` property instead...!